src/renderers/dom/shared/hooks/ReactDOMUnknownPropertyHook.js

```diff
@@ -76,54 +59,54 @@ if (__DEV__) {
       // Don't check events in this case.
       return true;
     }
-    warnedProperties[name] = true;
-    var lowerCasedName = name.toLowerCase();
-
-    // data-* attributes should be lowercase; suggest the lowercase version
-    var standardName = DOMProperty.isCustomAttribute(lowerCasedName)
-      ? lowerCasedName
-      : DOMProperty.getPossibleStandardName.hasOwnProperty(lowerCasedName)
-          ? DOMProperty.getPossibleStandardName[lowerCasedName]
-          : null;
 
     var registrationName = EventPluginRegistry.possibleRegistrationNames.hasOwnProperty(
       lowerCasedName,
     )
       ? EventPluginRegistry.possibleRegistrationNames[lowerCasedName]
       : null;
 
-    if (standardName != null) {
+    if (registrationName != null) {
       warning(
         false,
-        'Unknown DOM property %s. Did you mean %s?%s',
+        'Unknown event handler property %s. Did you mean `%s`?%s',
         name,
-        standardName,
+        registrationName,
         getStackAddendum(debugID),
       );
       return true;
-    } else if (registrationName != null) {
+    }
+
+    // data-* attributes should be lowercase; suggest the lowercase version
```

Comment:
It looks like, on master, React actually don't warn on cased data attributes. I believe this is because they get picked up by the custom attribute logic at the top:

https://github.com/facebook/react/blob/master/src/renderers/dom/shared/hooks/ReactDOMUnknownPropertyHook.js#L57-L58

Is it possible that this comment is out of date?